### ISSUE TITLE
Potential issue when `this.store[fetchMoreForQueryId]` is `undefined`

### ISSUE THREAD

**Issue by conrad-vanl** [apollographql/apollo-client#3345]
I'm seeing this error logged in a production app every so often:

```
undefined is not an object (evaluating 'this.store[o].networkStatus=t.NetworkStatus.ready')
```

I believe I've traced it to here:
https://github.com/apollographql/apollo-client/blob/6abcd727818398bff53bcd73e09a29b902557cc2/packages/apollo-client/src/data/queries.ts#L128-L130

I see above this function checks for the existence of `this.store[queryId]`, would it make sense to do a similar check for `this.store[fetchMoreForQueryId]` inside of this conditional? Happy to open a PR, if so.

Trying to think through what could be happening....a `fetchMore` is triggered, and then resolves after the component is unmounted perhaps? Not sure of how the lifecycle on the `graphql` HOC plays here.

**Comment by rtymchyk** [comment]
I noticed this as well. Does your app use SSR? The way I can reproduce it 100%:
* SSR a page, that on mount fires a `fetchMore` query
* Dismount that component before `fetchMore` completes (slow network)

**Comment by conrad-vanl** [comment]
No, this is a React native app. So doesn’t seem to be the case for us

**Comment by haani104** [comment]
![img_2437](https://user-images.githubusercontent.com/6011865/39085307-c96e8b08-45aa-11e8-83b3-93f822fcadeb.JPG)

**Comment by kdong007** [comment]
same

**Comment by hellogerard** [comment]
👋 Surfing in here from Google. To add more background to this for anyone else, we encountered this error as well, but only on some of our FlatLists. Then I figured out they differed in how we were working around this [bug in FlatList](https://github.com/facebook/react-native/issues/16067) which fired our `onEndReached` function multiples times on component unmount.  That bug causes FlatList to call `onEndReached` multiples times when the FlatList is unmounted. By avoiding these spurious calls in "user-land", it fixed the crash in this issue (also saving wasted network requests!).

So while I don't know _exactly_ what's going on in Apollo when a component is unmounted when multiple queries are in flight, I believe this is ~~the~~ a deeper, root cause.
